src/main/java/io/stargate/sgv2/jsonapi/service/operation/tasks/TaskRetryPolicy.java

```diff
@@ -0,0 +1,78 @@
+package io.stargate.sgv2.jsonapi.service.operation.tasks;
+
+import io.stargate.sgv2.jsonapi.util.PrettyPrintable;
+import io.stargate.sgv2.jsonapi.util.PrettyToStringBuilder;
+import java.time.Duration;
+import java.util.Objects;
+
+/**
+ * A policy for retrying in a {@link BaseTask}, this is used at the task level and so it in addition
+ * to any retrying a driver (like cassandra driver) may do.
+ *
+ * <p>If the attempt does not want to retry then it should use {@link TaskRetryPolicy#NO_RETRY}.
+ *
+ * <p>To implement a custom retry policy, subclass this class and override {@link
+ * #shouldRetry(Throwable)}.
+ */
+public class TaskRetryPolicy implements PrettyPrintable {
+
+  /** Re-usable policy to not retry. */
+  public static final TaskRetryPolicy NO_RETRY = new TaskRetryPolicy();
+
+  private final int maxRetries;
```

Comment:
Based on usage, isn't this more like "maxTries" than retries? Esp. when `NO_RETRY` is built with `maxRetries` of `1`?

Reply:
yes, leaving open to change in the next PR